### PR TITLE
Address::{is_ed25519_backed, backing_ed25519}

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/types/block/address/mod.rs [hunk 1]
```diff
@@ -113,6 +113,25 @@ impl Address {
...
+
+    /// Returns the backing [`Ed25519Address`], if any.
+    pub fn backing_ed25519(&self) -> Option<&Ed25519Address> {
+        match self {
+            Self::Ed25519(ed25519) => Some(ed25519),
+            Self::ImplicitAccountCreation(implicit) => Some(implicit.ed25519_address()),
+            Self::Restricted(restricted) => restricted.address().as_ed25519_opt(),
+            _ => None,
+        }
+    }
```
Comment (Alex6323) [inline sdk/src/types/block/address/mod.rs:133]: what about multi addresses? :thinking: (especially if it only contains a single weighted ed25519 address)

Comment (thibault-martinez) [reply sdk/src/types/block/address/mod.rs:133]: I actually think MultiAddress shouldn't be allowed to be a single address. I spoke with Phillip and he'll talk to the team to disallow or not. Anyway, I wouldn't really consider multi addresses here as there is no clear mapping to a single address being able to sign.

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — COMMENTED** [review]
👍 was checking if we could use this somewhere already but doesnt seem so easily

**Comment by thibault-martinez** [comment]
> 👍 was checking if we could use this somewhere already but doesnt seem so easily

I actually think we can but I'd do that when refactoring some parts.

**Pull request merged**